tests/test_response/test_room_response.py

```diff
@@ -155,3 +159,33 @@ def test_get_paginated_responses_empty_page(self):
             query_paginated_responses_empty_page,
             "Page does not exist"
         )
+
+    def test_room_response_with_date_range(self):
+        """
+        Testing for room response with date range
+        """
+        CommonTestCases.admin_token_assert_equal(
+            self,
+            get_room_response_query_by_date,
+            get_room_response_query_by_date_query
+        )
+
+    def test_room_response_with_invalid_date(self):
+        """
+        Testing for room response with invalid/ future dates
+        """
+        CommonTestCases.admin_token_assert_in(
+                self,
+                get_room_response_query_with_invalid_date,
+                "Dates should be before today"
+            )
+
+    def test_room_response_with_invalid_count(self):
```

Comment:
Good work @RachaelNantale, would this method name not be better written as `test_room_with_invalid_date_difference`?

Reply:
Oh yeah thanks for this @ThaDeveloper. This has been implemented